src/management/list.ts

```diff
@@ -34,6 +35,67 @@ export async function listFirebaseProjects(
   }
 }
 
+/**
+ * Send recurring API requests to list all Firebase apps of a Firebase project
+ * @return a promise that resolves to the list of all Firebase apps.
+ */
+export async function listFirebaseApps(
+  projectId: string,
+  platform?: AppPlatform,
```

Comment:
What happens if `platform` is undefined? Or when it's set to `PLATFORM_UNSPECIFIED`? Are these cases that should error before making a request?

Seems to me that `platform` should probably not be optional, and this should error if it's `UNSPECIFIED`. What do you think?

Reply:
If user does not pass `platform`, the command will list all apps, that's why `platform` can be `null`. According to the documentation of [FirebaseAppInfo](https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects/searchApps#FirebaseAppInfo), `PLATFORM_UNSPECIFIED` is an unknown state, which can happen for some apps (although rare). 

Some possible suggestions:
  - keep `platform` parameter nullable, and `undefined` indicates that we should list all apps
  - use `PLATFORM_UNSPECIFIED` as an indication to list all apps (seems not so right for me)
  - Just use `string` instead of `AppPlatform` across input parameters, and only use `AppPlatform` in `AppMetadata` from server response.

@nbegley That's what I understand about `PLATFORM_UNSPECIFIED`, can you confirm?

Reply:
> According to the documentation of [FirebaseAppInfo](https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects/searchApps#FirebaseAppInfo), `PLATFORM_UNSPECIFIED` is an unknown state, which can happen for some apps (although rare).

A small clarification here - the docs say that `PLATFORM_UNSPECIFIED` is an "Unknown state. This is only used for distinguishing unset values". So if PLATFORM_UNSPECIFIED is ever returned by Firebase's API, that would be a bug, since every app has to have a platform.

> Some possible suggestions:
> 
> * keep `platform` parameter nullable, and `undefined` indicates that we should list all apps

This is what you currently have implemented, and it seems reasonable to me.

> * use `PLATFORM_UNSPECIFIED` as an indication to list all apps (seems not so right for me)

This is also reasonable, but I agree with you that I think your current solution is more correct.

> * Just use `string` instead of `AppPlatform` across input parameters, and only use `AppPlatform` in `AppMetadata` from server response.

-1 to this option.

Reply:
Keeping option 1 for now, @bkendall  your opinion?